### ISSUE TITLE
Tabulator.defaultOptions.columnDefaults vs. table's columnDefaults

### ISSUE THREAD

**Issue by 500Foods** [olifolkerd/tabulator#4010]
**Website Page**
Not sure if this is a bug, feature, or documentation issue, but let's go with documentation.
https://tabulator.info/docs/5.4/modules#default

**Describe the issue**
I routinely use the Tabulator.defaultOptions mechanism to set "global defaults" for Tabulator settings.  For example, headerSortTristate = true, a feature I'm particularly fond of.  This makes it easy to ensure that anytime I have a sortable column visible, it is a tristate sort.  All good!

However, I recently got into the habit of adding a columnDefaults section to individual tables, and I noticed that my tristate option was no longer set.  I'm guessing this is because the columnDefaults set in the table replaces the global value of columnDefaults, rather than augmenting it as one might have expected?  

While I might prefer that mode of operation, It should likely be made clear in the documentation that this is how it works.

**Comment by olifolkerd** [comment]
Yes that is correct, anything other than replacing the option outright becomes complex because with nested objects it becomes unclear as to wether at each level they should be merged or overidden and there can be further challenges around arrays, or properties that can handle multiple types.

I'm certainly happy to look at adding multiple modes for how the default options can be employed in the future.

But for now I'm happy to look at an update to the docs.

**Comment by 500Foods** [comment]
Every once in a while I find myself skimming through the docs for something and then end up learning something else entirely or seeing something in a new light.  For example, I've been using the Tabulator.defaultOptions since I first heard of them, very early in my Tabulator adventures, but only yesterday did I try using columnDefaults for a table individually.  Hence running across this issue.  Not a big one of course, but noticed that this wasn't obvious from the docs.  I'd have thought that it would apply columnDefaults over top of whatever was done previously, but that's just how my brain works.  As you say, things are never as simple as they might seem!

**Comment by olifolkerd** [comment]
Hey @500Foods 

I have updated the docs on default options to make the functionality clearer and will tag this issue as a feature request to look at adding the merge option for handling object values

Cheers

Oli :)
